cachi2/core/package_managers/pip.py

```diff
@@ -792,7 +781,7 @@ class PipRequirementsFile:
     def __init__(self, file_path):
         """Initialize a PipRequirementsFile.
 
-        :param str file_path: the full path to the requirements file
+        :param file_path: the full path to the requirements file
```

Comment:
Apologies for not adding type hints to the PipRequirement and PipRequirementsFile classes. Those two (and any methods that use them) cannot possibly pass type-checking without a larger refactoring :cry:

Reply:
Just to illustrate some of the problems:


PipRequirementsFile initialized with file_path=None:

https://github.com/containerbuildsystem/cachi2/blob/36d398e1d57819d819fd051ee78f611e628f4bb9/cachi2/core/package_managers/pip.py#L808

PipRequirementsFile._read_lines doesn't work if file path is None:

https://github.com/containerbuildsystem/cachi2/blob/36d398e1d57819d819fd051ee78f611e628f4bb9/cachi2/core/package_managers/pip.py#L872

Everything in PipRequirement initialized as None:

https://github.com/containerbuildsystem/cachi2/blob/36d398e1d57819d819fd051ee78f611e628f4bb9/cachi2/core/package_managers/pip.py#L981-L999

None of the PipRequirement methods work if those attributes are None.